internal/natsclient/jetstream.go

```diff
@@ -0,0 +1,157 @@
+package natsclient
+
+import (
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"go.uber.org/zap"
+)
+
+type Message struct {
+	Subject string
+	Data    []byte
+}
+
+// Jetstream represents the NATS core handler
+type Jetstream struct {
+	connection *nats.Conn
+	jetstream  nats.JetStreamContext
+	config     *Config
+	logger     *zap.Logger
+	metrics    Metrics
+}
+
+// NewJetstream initializes NATS JetStream connection
+func NewJetstream(config Config, logger *zap.Logger) *Jetstream {
+	j := Jetstream{
+		config:  &config,
+		logger:  logger,
+		metrics: NewMetrics(),
+	}
+
+	j.connect()
+
+	j.createJetstreamContext()
+
+	j.createStream()
+
+	return &j
+}
+
+func (j *Jetstream) connect() {
+	var err error
+	j.connection, err = nats.Connect(j.config.URL)
+	if err != nil {
+		j.logger.Panic("could not connect to nats", zap.Error(err))
+	}
+
+	j.connection.SetDisconnectErrHandler(func(_ *nats.Conn, err error) {
+		j.metrics.ConnectionErrors.Add(1)
+		j.logger.Error("nats disconnected", zap.Error(err))
+	})
+
+	j.connection.SetReconnectHandler(func(_ *nats.Conn) {
+		j.logger.Warn("nats reconnected")
+	})
+}
+
+func (j *Jetstream) createJetstreamContext() {
+	var err error
+	j.jetstream, err = j.connection.JetStream()
+	if err != nil {
+		j.logger.Panic("could not connect to jetstream", zap.Error(err))
+	}
+}
+
+func (j *Jetstream) createStream() {
+	_, err := j.jetstream.AddStream(&nats.StreamConfig{
+		Name:     "test",
+		Subjects: []string{"test.*"},
+	})
+	if err != nil {
+		j.logger.Panic("could not add stream", zap.Error(err))
+	}
+}
+
+func (j *Jetstream) StartJetstreamMessaging() {
```

Comment:
i suggest renaming this method to sth like startblackboxtest or sth.

Reply:
Fixed. 👍